tests/test_crawlers.py

```diff
@@ -561,90 +561,98 @@ def test_get_download_url_no_direct_download_link(self, mock_get_link, mock_http
         self.assertIsNone(crawlers.ThreddsCrawler('').get_download_url("dummy"))
 
 
-class CopernicusOpenSearchAPICrawlerTestCase(unittest.TestCase):
-    """Tests for the Copernicus OpenSearch API crawler"""
-    BASE_URL = 'https://scihub.copernicus.eu/dhus/search'
-    SEARCH_TERMS = '(platformname:Sentinel-1 OR platformname:Sentinel-3) AND NOT L0'
-    PAGE_SIZE = 2
-    TEST_DATA = {
-        'page1': {
-            'offset': 0,
-            'file_path': "data/copernicus_opensearch/page1.xml"},
-        'page2': {
-            'offset': 2,
-            'file_path': "data/copernicus_opensearch/page2.xml"},
-        'page3': {
-            'offset': 4,
-            'file_path': 'data/copernicus_opensearch/page3.xml'}
-    }
+class HTTPPaginatedAPICrawlerTestCase(unittest.TestCase):
+    """Tests for the HTTPPaginatedAPICrawler base class"""
+
+    class TestCrawler(crawlers.HTTPPaginatedAPICrawler):
```

Comment:
A class inside a class must be in the setup method in order to be fresh and unchanged of execution of the other tests.

Reply:
This is a class definition, not an instantiation.
It is instantiated in the `setUp()` method.